packages/orm/src/loadHints.ts

```diff
@@ -38,16 +40,18 @@ export type MarkLoaded<T extends Entity, P, UH = {}> =
       ? LoadedReference<T, Loaded<U, UH>, N>
       : P extends Collection<MaybeBaseType, infer U>
         ? LoadedCollection<T, Loaded<U, UH>>
-        : P extends AsyncProperty<MaybeBaseType, infer V>
-          ? // prettier-ignore
-            [V] extends [(infer U extends Entity) | undefined]
+        : P extends ReadOnlyCollection<MaybeBaseType, infer U>
+          ? LoadedReadOnlyCollection<T, Loaded<U, UH>>
```

Comment:
The indentation makes it look like a bigger change, but just added `ReadOnlyCollection` checking ... at some point, it would be neat to collapse this huge ternary into something cleaner, i.e. if the `P extends ReadOnlyCollection` could become something like:

```
P extends Loadable<..., infer L> ? L
```

Where each Loadable type could have a generic that pointed to its Loaded counterpart...